sim/battle.js

```diff
@@ -1791,7 +1791,12 @@ class Battle extends Dex.ModdedDex {
 		defense = this.runEvent('Modify' + statTable[defenseStat], defender, attacker, move, defense);
 
 		//int(int(int(2 * L / 5 + 2) * A * P / D) / 50);
-		let baseDamage = Math.floor(Math.floor(Math.floor(2 * level / 5 + 2) * basePower * attack / defense) / 50);
+		let baseDamage;
+		if (this.gen == 4) {
```

Comment:
It shouldn't, since it's (this,gen == 4) rather than (this.gen <= 4) or (this.gen >= 4)